src/net.h

```diff
@@ -653,7 +653,8 @@ class CNode
     // Address of this peer
     const CAddress addr;
     // Bind address of our side of the connection
-    const CAddress addrBind;
+    // Non-const to allow lazy initialization if getsockname fails initially
+    CAddress addrBind;
```

Comment:
If we remove the const, wouldn't that be undesirable as it could break parts of the codebase that rely on this assumption?

Can you identify why getsockname is failing?  CAddress is not a simple POD type and may therefore be in an inconsistent state if read during a partial update.